Userland/Services/RequestServer/ConnectionFromClient.cpp

```diff
@@ -443,8 +449,38 @@ void ConnectionFromClient::ensure_connection(URL::URL const& url, ::RequestServe
         return;
     }
 
-    (void)cache_level;
-    dbgln("FIXME: EnsureConnection: Pre-connect to {}", url);
+    auto const url_string_value = url.to_string().value();
+
+    if (cache_level == CacheLevel::CreateConnection) {
+        auto* easy = curl_easy_init();
+        if (!easy) {
+            dbgln("EnsureConnection: Failed to initialize curl easy handle");
+            return;
+        }
+
+        auto set_option = [easy](auto option, auto value) {
+            auto result = curl_easy_setopt(easy, option, value);
+            if (result != CURLE_OK) {
+                dbgln("EnsureConnection: Failed to set curl option: {}", curl_easy_strerror(result));
+                return false;
+            }
+            return true;
+        };
+
+        set_option(CURLOPT_URL, url_string_value.to_byte_string().characters());
+        set_option(CURLOPT_PORT, url.port_or_default());
+        set_option(CURLOPT_CONNECTTIMEOUT, s_connect_timeout_seconds);
+        set_option(CURLOPT_CONNECT_ONLY, 1L);
+        set_option(CURLOPT_SHARE, m_curl_share);
+
+        curl_easy_perform(easy);
```

Comment:
:thinking: I assume this will block RS, is there a way to let curl do this...later?
maybe adding it to the curl_multi thingy with a sentinel activerequest and handling that in the appropriate callbacks?